### PR TITLE
support 'diff-only' hooks by passing diff via stdin (#1279)

### PR DESCRIPTION
**Pull request by koreno**
this is a poc/draft, with the intention to make minimal changes and see if this can work

### CODE REVIEW

File: pre_commit/commands/run.py [hunk 1]
```diff
@@ -117,9 +117,14 @@ def _run_single_hook(classifier, hook, skips, cols, verbose, use_color):
 
         diff_cmd = ('git', 'diff', '--no-ext-diff')
         diff_before = cmd_output_b(*diff_cmd, retcode=None)
-        filenames = tuple(filenames) if hook.pass_filenames else ()
+        if hook.pass_diff:
+            inputs = (diff_before[1],)
```
Comment (asottile) [inline pre_commit/commands/run.py]: this isn't the correct value fwiw, `diff_before` is the *unstaged* diff (used to determine whether things change or not)

**Comment by asottile** [comment]
if it wasn't clear, I don't want this change -- sorry

**Comment by koreno** [comment]
I wrote the code before I got your response.
Thanks anyway, good luck.

**Comment by asottile** [comment]
via #1279

**Pull request closed without merging**